src/components/AdditionalProfileInfo/AdditionalProfileInfo.tsx

```diff
@@ -69,37 +88,41 @@ const AdditionalProfileInfo = ({ data, isLoading, updateCaptureOption }) => {
         value: setting.value
       };
     });
-    updateCaptureAnswers({
-      customAnswers
-    })
-      .then(() => {
-        for (let i = 0; i < customSettings.length; i += 1) {
-          updateCaptureOption({
-            key: customSettings[i].key,
-            value: customSettings[i].value
-          });
-        }
-        setMessage({
-          message: t(
-            'additional-profile-info.success-message',
-            'Your answers have been changed successfully'
-          ),
-          type: 'success'
-        });
-        setIsPending(false);
-        setIsSectionDisabled(true);
-      })
-      .catch(() => {
-        setMessage({
-          message: t(
-            'additional-profile-info.error.submit',
-            'Something went wrong. Try again later.'
-          ),
-          type: 'error'
-        });
-        setIsPending(false);
-        setIsSectionDisabled(true);
+
+    try {
+      await updateCaptureAnswers({
+        customAnswers
+      });
+
+      customSettings.forEach((setting) =>
```

Comment:
maybe it would make sense to have a bulk update action? what do you think? it would be just one operation instead of multiple ones